java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -2856,11 +2962,16 @@ public static TypeReader createTreeReader(TypeDescription readerType,
     }
   }
 
+  public static boolean isDecimalAsLong(OrcFile.Version version, int precision) {
```

Comment:
This could be part of another PR tbh

Reply:
I need this for the next PR conversion of SArg to Filter. I can remove this here and include it as part of that PR

Reply:
Sounds good! The SArgs PR will be smaller anyway so better to add there

Reply:
reverted this change